mtp_api/apps/security/tests/test_checks.py

```diff
@@ -557,3 +557,141 @@ def test_pending_check_created_for_monitored_user(self):
         self.assertEqual(payment.credit.resolution, CREDIT_RESOLUTION.INITIAL)
         self.assertTrue(hasattr(payment.credit, 'security_check'))
         self.assertEqual(payment.credit.security_check.status, CHECK_STATUS.PENDING)
+
+
+class AutoAcceptRuleTestCase(APITestCase, AuthTestCaseMixin):
+    fixtures = ['initial_types.json', 'test_prisons.json', 'initial_groups.json']
+
+    def setUp(self):
+        super().setUp()
+        self.users = make_test_users(clerks_per_prison=1)
+        prisoner_locations = load_random_prisoner_locations(number_of_prisoners=1)
+        generate_payments(payment_batch=1)
+        prisoner_profiles = generate_prisoner_profiles_from_prisoner_locations(prisoner_locations)
+        sender_profiles = generate_sender_profiles_from_payments(number_of_senders=1, reassign_dcsd=True)
+        prisoner_profiles[0].monitoring_users.add(self.users['security_fiu_users'][0].id)
+        sender_profiles[0].debit_card_details.first().monitoring_users.add(self.users['security_fiu_users'][0].id)
+
+        response = self.client.post(
+            reverse(
+                'security-check-auto-accept-list'
+            ),
+            data={
+                'prisoner_profile': prisoner_profiles[0].id,
+                'debit_card_sender_details': sender_profiles[0].debit_card_details.first().id,
+                'states': [
+                    {
+                        'reason': 'This person has amazing hair',
+                        'added_by_id': self.users['security_fiu_users'][0].id
```

Comment:
Does the client still needs to pass `states.added_by_id` if in the serializer [we set it from the context's `request.user`](https://github.com/ministryofjustice/money-to-prisoners-api/pull/626/files#diff-3cf42ec26dec1e69b6d44a59bb6095460400d603136ea39c825fd17666fa2a86R393)? e.g. is `states.added_by_id` a mandatory field?

Maybe it's because of how serializers work.

Reply:
the field on the serialiser is `added_by` and i'm not sure a related model field would be magical enough to also try a `added_by_id` param in the data. i suspect this line has no effect.

Reply:
So this is an interesting one, I believe this behavior actually changed in Django 2.2, which we've recently upgraded to. Now when `ModelSerializer` serializes a model with a foreign key and no explicit nested serializer defined, it strips off the '_id' suffix of the field. I think it's backwards compatible on deserialization, but not on serialization (i.e. it will accept the previous behavior of the `_id` suffix, but won't add it when serializing it and any other objects to the response.

I'll rename the `added_by_id` key,  to `added_by` for consistency though, as that's what the actual call that's being made attaches as a parameter.

Reply:
oh ok. still, you shouldn't need to explicitly pass it in as noms-ops doesn't and the serialiser automatically gets it from the request